flake-modules/dev/new-plugin.py

```diff
@@ -0,0 +1,79 @@
+#!/usr/bin/env python3
+
+import os
+import re
+from argparse import ArgumentParser
+
+# Template for default.nix
+default_nix_template = """{{ lib, ... }}:
+lib.nixvim.neovim-plugin.mkNeovimPlugin {{
+  name = "{name}";
+  originalName = "{originalName}";
+  package = "{package}";
+
+  maintainers = [ lib.maintainers.YOUR_NAME ];
+
+  settingsOptions = {{
+    exampleOption = lib.nixvim.defaultNullOpts.mkBool false ''
+      Example option for the {name} plugin.
+    '';
+  }};
+}}
+"""
+
+# Template for test file
+test_nix_template = """{{
+  empty = {{
+    plugins.{name}.enable = true;
+  }};
+}}
+"""
+
+
+def snake_case(input_string):
+    # Convert to snake_case and remove trailing underscores
+    return re.sub(r"[\W_]+", "_", input_string).lower().replace("nvim", "").rstrip("_")
```

Comment:
Feels like this may benefit from some unit tests.

One edge case, for example: how does it handle leading `nvim-` prefixes?

Also, don't we normally use kebab-case for plugin names?

Reply:
Yeah, late night brain fart lol.